src/test/java/org/opentripplanner/routing/fares/OrcaFareServiceTest.java

```diff
@@ -0,0 +1,162 @@
+        orcaFareService.IS_TEST = true;
+        DEFAULT_RIDE_PRICE_IN_CENTS = OrcaFareServiceImpl.DEFAULT_TEST_RIDE_PRICE * 100;
+    }
+
+    /**
+     * These tests are designed to specifically validate Orca fares. Since these fares are hard-coded, it is acceptable
+     * to make direct calls to the Orca fare service with predefined routes. Where the default fare is applied a test
+     * substitute {@link OrcaFareServiceImpl#DEFAULT_TEST_RIDE_PRICE} is used. This will be the same for all cash fare
+     * types.
+     */
+    private static void calculateFare(List<Ride> rides,
+                                  Fare.FareType fareType,
+                                  float expectedFareInCents
+    ) {
+        Fare fare = new Fare();
+        orcaFareService.populateFare(fare, null, fareType, rides, null);
+        Assertions.assertEquals(expectedFareInCents, fare.getFare(fareType).getCents());
+    }
+
+    /**
+     * Test to confirm the correct transfer cost per fare type within a single agency.
+     */
+    @Test
+    public void calculateFareForSingleAgency() {
+        Ride ride = getRide(COMM_TRANS_AGENCY_ID, "400", 0);
+        List<Ride> rides = Lists.newArrayList(ride);
+        calculateFare(rides, Fare.FareType.regular, DEFAULT_RIDE_PRICE_IN_CENTS);
+        calculateFare(rides, Fare.FareType.senior, DEFAULT_RIDE_PRICE_IN_CENTS);
+        calculateFare(rides, Fare.FareType.youth, DEFAULT_RIDE_PRICE_IN_CENTS);
+        calculateFare(rides, Fare.FareType.electronicSpecial, 200f);
+        calculateFare(rides, Fare.FareType.electronicRegular, DEFAULT_RIDE_PRICE_IN_CENTS);
+        calculateFare(rides, Fare.FareType.electronicSenior, 200f);
+        calculateFare(rides, Fare.FareType.electronicYouth, 300f);
+    }
+
+    /**
+     * WSF do not except free transfers. This test is to make sure the rider is charged the cash price for WSF as well
+     * as the highest fare where Orca can be used.
+     */
+    @Test
+    public void calculateFareWithNoFreeTransfer() {
+        List<Ride> rides = Lists.newArrayList();
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 0));
+        rides.add(getRide(WASHINGTON_STATE_FERRIES_AGENCY_ID, 1));
+        rides.add(getRide(COMM_TRANS_AGENCY_ID, 2));
+        calculateFare(rides, Fare.FareType.regular, DEFAULT_RIDE_PRICE_IN_CENTS * 3);
+        calculateFare(rides, Fare.FareType.senior, DEFAULT_RIDE_PRICE_IN_CENTS * 3);
+        calculateFare(rides, Fare.FareType.youth, DEFAULT_RIDE_PRICE_IN_CENTS * 3);
+        calculateFare(rides, Fare.FareType.electronicSpecial, 0f + DEFAULT_RIDE_PRICE_IN_CENTS + 125f);
+        calculateFare(rides, Fare.FareType.electronicRegular, 0f + DEFAULT_RIDE_PRICE_IN_CENTS + DEFAULT_RIDE_PRICE_IN_CENTS);
+        calculateFare(rides, Fare.FareType.electronicSenior, 0f + DEFAULT_RIDE_PRICE_IN_CENTS + 125f);
+        calculateFare(rides, Fare.FareType.electronicYouth, 200f + DEFAULT_RIDE_PRICE_IN_CENTS + 0f);
```

Comment:
I'm pretty sure the youth reduced fare for Kitsap is $1. See [here](https://www.kitsaptransit.com/fares/fares).